src/librawspeed/decoders/SrwDecoder.cpp

```diff
@@ -52,21 +52,15 @@ RawImage SrwDecoder::decodeRawInternal() {
 
   if (32769 == compression)
   {
-    bool bit_order = false;  // Default guess
-    auto msb_hint = hints.find("msb_override");
-    if (msb_hint != hints.end())
-      bit_order = ("true" == (msb_hint->second));
+    bool bit_order = hints.get("msb_override", false);
     this->decodeUncompressed(raw, bit_order ? BitOrder_Jpeg : BitOrder_Plain);
     return mRaw;
   }
 
   if (32770 == compression)
   {
     if (!raw->hasEntry ((TiffTag)40976)) {
-      bool bit_order = (bits == 12);  // Default guess
-      auto msb_hint = hints.find("msb_override");
-      if (msb_hint != hints.end())
-        bit_order = ("true" == (msb_hint->second));
```

Comment:
So, i checked, and there are three hints with boolean values:
```
$ grep -i "hint " cameras.xml | sort | uniq | grep "true\|false"
                        <Hint name="msb_override" value="false"/>
                        <Hint name="nikon_wb_adjustment" value="true"/>
                        <Hint name="wb_mangle" value="true"/>
```
And apparently the `msb_override` can also have a value of `` (empty). With old code that would mean false.
```
$ grep -i "hint " cameras.xml | sort | uniq | grep "msb_override\|nikon_wb_adjustment\|wb_mangle"
                        <Hint name="msb_override" value=""/>
                        <Hint name="msb_override" value="false"/>
                        <Hint name="nikon_wb_adjustment" value="true"/>
                        <Hint name="wb_mangle" value="true"/>
```
I guess to dispel my doubts i will just have to write a test ))